### ISSUE TITLE
Reproduction of results

### ISSUE THREAD

**Issue by pipizhum** [xulianuwa/MCTformer#37]
Can you provide instructions for all the training steps? I followed the author's instructions and the final segmentation result of mIoU was only 67.5%.

**Comment by smart0eddie** [comment]
I also have problem reproducing the resuts following the given .sh files
How to evaluate the seed and mask performance?
I got 57.3% with MCTformer V2 and 61.7% after post processing with psa on VOC train-aug

**Comment by smart0eddie** [comment]
May you please explain which dataset is used for each step of the training?

According to the run.sh, run_psa.sh, and run_seg.sh,
1. VOC12_train_aug is used to train MCTformerV2
2. VOC12_train_aug is used to geenrate attention map (the seed?)
3. VOC12_train_aug is used to train PSA model
4. VOC12_train is used to generate the pseudo label mask
5. VOC12_train is used to evaluate the pseudo label mask
6. VOC12_train_aug is used to train the segmentation model

Why only evaluate the VOC12_train instead of the extended VOC12_train_aug if using the whole VOC12_train_aug to train the models?

**Comment by xulianuwa** [comment]
> May you please explain which dataset is used for each step of the training?
> 
> According to the run.sh, run_psa.sh, and run_seg.sh,
> 
> 1. VOC12_train_aug is used to train MCTformerV2
> 2. VOC12_train_aug is used to geenrate attention map (the seed?)
> 3. VOC12_train_aug is used to train PSA model
> 4. VOC12_train is used to generate the pseudo label mask
> 5. VOC12_train is used to evaluate the pseudo label mask
> 6. VOC12_train_aug is used to train the segmentation model
> 
> Why only evaluate the VOC12_train instead of the extended VOC12_train_aug if using the whole VOC12_train_aug to train the models?

The seed & pseudo mask evaluation steps are performed on the original training set (1464 images), and the training steps use the augmented training set (10582 images), which is a common practice.

**Comment by Italy2006** [comment]
> Can you provide instructions for all the training steps? I followed the author's instructions and the final segmentation result of mIoU was only 67.5%.

Have you solved this problem? I get the same result

**Comment by Italy2006** [comment]
> Edit: when evaluating on the VOC train instead of VOC train_aug, I get the performance close to the ones in the paper (61.5 and 68.0 compared to 61.7 and 69.1) However, the segmentation mIoU is only 69.8 compared to 71.9 in the paper
> 
> I also have problem reproducing the resuts following the given .sh files How to evaluate the seed and mask performance? I got 57.3% with MCTformer V2 and 61.7% after post processing with psa on VOC train-aug

Have you solved this problem? I get the same result!

**Comment by smart0eddie** [comment]
I think it's due to the random walk post processing

**Comment by Italy2006** [comment]
> I think it's due to the random walk post processing

I totally follow the author, but the pesudo label only achieve 68.3%, and the seg results only achieve 66.7%(single scale), have you solved it?

**Comment by duan-song** [comment]
> Can you provide instructions for all the training steps? I followed the author's instructions and the final segmentation result of mIoU was only 67.5%.

I have the same problem. But, I use RTX 4090 to train the MCFformer+ with 16 batch size, which maybe cause this problem. What GPU and batch size are you using?

**Comment by YanGe1105** [comment]
> Can you provide instructions for all the training steps? I followed the author's instructions and the final segmentation result of mIoU was only 67.5%.

I also obtained similar results. Do you also use MCTformer+ with Transformer model?

**Comment by Italy2006** [comment]
No，i can‘t reproduce the result，so i choose another code



发自我的iPhone


------------------ Original ------------------
From: YanGe1105 ***@***.***&gt;
Date: Mon,Dec 30,2024 4:45 PM
To: xulianuwa/MCTformer ***@***.***&gt;
Cc: Italy2006 ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [xulianuwa/MCTformer] Reproduction of results (Issue #37)





  
Can you provide instructions for all the training steps? I followed the author's instructions and the final segmentation result of mIoU was only 67.5%.
  
I also obtained similar results. Do you also use MCTformer+ with Transformer model?
